src/Mvc/test/WebSites/BasicWebSite/Controllers/CustomValueProviderController.cs

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using Microsoft.AspNetCore.Mvc;
+
+namespace BasicWebSite.Controllers;
+
+public class CustomValueProviderController : Controller
+{
+    [HttpGet]
+    public string CustomValueProviderDisplayName(string customValueProviderDisplayName)
+        => customValueProviderDisplayName;
+
+    [HttpGet]
+    public int[] CustomValueProviderIntValues(int[] customValueProviderIntValues)
```

Comment:
Could you please add another test for a `int?` parameter? I know that it might be redundant but at least we safe guard any changes to nullable behaviors.

Based on you change the test should result in `null,42,100,null,200`

Reply:
``` c#
[HttpGet]
    public int?[] CustomValueProviderNullableIntValues(int?[] customValueProviderIntValues)
        => customValueProviderIntValues;
```